src/components/timers/Tabata.tsx

```diff
@@ -1,3 +1,214 @@
+      } else {
+        intervalId = setInterval(() => {
+          setTime(prevTime => {
+            const newTime = prevTime + 10;
+            if ((isWorking && newTime >= workTime * 1000) || (!isWorking && newTime >= restTime * 1000)) {
+              if (isWorking) {
+                setIsWorking(false);
+                setCountdown(3);
+                return 0;
+              } else {
+                if (round < rounds) {
+                  setRound(prevRound => prevRound + 1);
+                  setIsWorking(true);
+                  setCountdown(3);
+                  return 0;
+                } else {
+                  setIsRunning(false);
+                  return newTime;
+                }
+              }
+            }
+            return newTime;
+          });
+        }, 10);
+      }
+    }
+    return () => clearInterval(intervalId);
+  }, [isRunning, isWorking, workTime, restTime, round, rounds, countdown]);
+
+  const handleStartStop = () => {
+    if (isConfiguring) {
+      setIsConfiguring(false);
+      setTime(0);
+      setCountdown(3);
+    }
+    setIsRunning(!isRunning);
+  };
+
+  const handleReset = () => {
+    setTime(0);
+    setRound(1);
+    setIsWorking(true);
+    setIsRunning(false);
+    setIsConfiguring(true);
+    setCountdown(3);
+  };
+
+  const handleEnd = () => {
+    setIsRunning(false);
+    setTime(workTime * 1000);
+    setRound(rounds);
+    setIsWorking(false);
+    setCountdown(3);
+  };
+
+  const formatTime = (timeInMilliseconds: number): string => {
+    const minutes = Math.floor(timeInMilliseconds / 60000);
+    const seconds = Math.floor((timeInMilliseconds % 60000) / 1000);
+    const milliseconds = Math.floor((timeInMilliseconds % 1000) / 10);
+    const microseconds = timeInMilliseconds % 10;
+    return `${minutes.toString().padStart(2, '0')}:${seconds.toString().padStart(2, '0')}.${milliseconds.toString().padStart(2, '0')}${microseconds}`;
+  };
+
+  return (
+    <div style={styles.container}>
+      {isConfiguring ? (
+        <div>
+          <div>
+            Work Time (seconds): 
+            <input 
+              style={styles.input}
+              type="number" 
+              value={workTime} 
+              onChange={(e) => setWorkTime(Math.max(1, parseInt(e.target.value) || 0))} 
+            />
+          </div>
+          <div>
+            Rest Time (seconds): 
+            <input 
+              style={styles.input}
+              type="number" 
+              value={restTime} 
+              onChange={(e) => setRestTime(Math.max(1, parseInt(e.target.value) || 0))} 
+            />
+          </div>
+          <div>
+            Number of Rounds: 
+            <input 
+              style={styles.input}
+              type="number" 
+              value={rounds} 
+              onChange={(e) => setRounds(Math.max(1, parseInt(e.target.value) || 0))} 
+            />
+          </div>
+        </div>
+      ) : (
+        <>
+          {countdown > 0 ? (
```

Comment:
I like this countdown functionality! One thing to watch out for - if I click `end` while the countdown is still running, it freezes in place. You should either clear the countdown and show the end state, or prevent users from pressing `end` during countdown.